### PR TITLE
Feature: Doom Gaze can now reward a character

### PR DESCRIPTION
**Pull request by kielbasiago**
Built on top of the force-item-reward-checks branch. 

Here's the full feature in action https://youtu.be/4jncZut2ieQ

The dance afterwards was added specifically to account for invisible dialogs after you recruit the character. We force the fade in and then do the animation, which will look much  better if there's no objective reward.  

**Also fixed some character gating bugs - Dragon characters would never have characters behind them, and now they can.**

### CODE REVIEW

File: constants/check_presets.py [hunk 1]
```diff
@@ -33,12 +34,30 @@
     ],
 )
 
-NO_FREE_PROGRESSION_NEW = CheckPreset(
+NO_FREE_CHARACTERS = CheckPreset(
```
Comment (kielbasiago) [inline constants/check_presets.py:37]: Added "No Free Characters" preset that Elastoid floated to me a while back

File: constants/checks.py [hunk 2]
```diff
@@ -1,76 +1,85 @@
 from collections import namedtuple
+from data.characters import Characters
 import data.event_bit as event_bit
 import data.npc_bit as npc_bit
 from event.event_reward import RewardType
 
-NameBit = namedtuple("NameBit", ["name", "bit", "reward_types"])
+NameBit = namedtuple("NameBit", ["name", "bit", "reward_types", "gate_character"])
```
Comment (kielbasiago) [inline constants/checks.py:7]: Adding gated character to each check, rather than tying it to each event module

Comment (kielbasiago) [reply constants/checks.py:7]: The reason for this was to allow 2/8 dragons to be gated behind character (Mt Zozo => Cyan, Ancient Castle => Edgar)

File: constants/check_presets.py [hunk 3]
```diff
@@ -48,17 +67,16 @@
         NARSHE_WEAPON_SHOP_MINES,
         SEALED_GATE,
         SOUTH_FIGARO_PRISONER,
+        TZEN_THIEF,
     ]
 )
 
-NO_FREE_PROGRESSION_CLASSIC = CheckPreset(
```
Comment (asilverthorn) [inline constants/check_presets.py:54]:
```
File "WorldsCollide\args\advanced_checks.py", line 43, in process
    from constants.check_presets import key_preset, NO_FREE_PROGRESSION_CLASSIC
ImportError: cannot import name 'NO_FREE_PROGRESSION_CLASSIC' from 'constants.check_presets'
```

Comment (asilverthorn) [reply constants/check_presets.py:54]: Fixed on beta: https://github.com/asilverthorn/WorldsCollide/commit/8c5685fbf8a141ea25fcaa96b16b2d770c5f8a01

**Pull request merged**